e2e/app/aou-elements/base-element.ts

```diff
@@ -0,0 +1,229 @@
+    const disabled = await this.getProperty('disabled');
+    return !!disabled;
+  }
+
+  /**
+   * <pre>
+   *  Check if the element is visible
+   * </pre>
+   * @param {Page} page
+   * @param {ElementHandle} element
+   */
+  async isVisible(): Promise<boolean> {
+    const boxModel = await this.element.boxModel();
+    return boxModel !== null;
+  }
+
+  /**
+   * Check both boxModel and style for visibility.
+   */
+  async isDisplayed() {
+    const isVisibleHandle = await this.page.evaluateHandle((e) =>
+    {
+      const style = window.getComputedStyle(e);
+      return (style && style.display !== 'none' &&
+         style.visibility !== 'hidden' && style.opacity !== '0');
+    }, this.element);
+    const jValue = await isVisibleHandle.jsonValue();
+    const boxModelValue = await this.element.boxModel();
+    if (jValue && boxModelValue) {
+      return true;
+    }
+    return false;
+  }
+
+  async click(options?: ClickOptions): Promise<void> {
+    await this.element.asElement().click(options);
+  }
+
+  async type(text: string, options?: { delay: number }): Promise<void> {
+    await this.focus();
+    await this.element.asElement().type(text, options);
+  }
+
+  async pressKeyboard(key: string, options?: { text?: string, delay?: number }): Promise<void> {
+    await this.element.asElement().press(key, options);
+  }
+
+  /**
+   * Calling focus() and hover() together.
+   */
+  async focus(): Promise<void> {
+    const handle = this.element.asElement();
+    await Promise.all([
+      handle.focus(),
+      handle.hover()
+    ]);
+  }
+
+  /**
+   * <pre>
+   * Get the textContent property value for a element.
+   * </pre>
+   */
+  async getTextContent(): Promise<string> {
+    const handle = await this.element.asElement();
+    return await handle.evaluate(
+       (element: HTMLElement) => (element.textContent ? element.textContent.trim() : ''), this.element,
+    );
+  }
+
+  /**
+   * Get the value of property 'value' for this element.
+   * Alternative: await page.evaluate(elem => elem.value, element);
+   */
+  async getValue(): Promise<unknown> {
+    return await this.getProperty('value');
+  }
+
+  async getComputedStyle(styleName: string): Promise<unknown> {
+    const handle = this.element.asElement();
+    const attrStyle = await handle.evaluateHandle((e) => {
+      const style = window.getComputedStyle(e);
+      return style;
+    }, this.element);
+
+    return (await attrStyle.getProperty(styleName)).jsonValue()
+  }
+
+  /**
+   * Determine if cursor is disabled (= not-allowed) by checking style attribute 'cursor'.
+   */
+  async isCursorNotAllowed(): Promise<boolean> {
+    const cursor = await this.getComputedStyle('cursor');
+    return cursor === 'not-allowed';
+  }
+
+  /**
+   * Finds element's size.
+   */
+  async size(): Promise<{ width: number; height: number }> {
+    const box = await this.element.boundingBox();
+    if (!box) { return { width: 0, height: 0 }; }
```

Comment:
opt_nit: In general I would not recommend 1-line if statements AND brackets. I'd do one or the other (usually in the codebase we use:
```
 if (...) {
 ...
}
```

Reply:
👍